File: push/src/main/java/com/nihility/service/RegisterRecordAbility.java
package com.nihility.service;

import android.content.Intent;

public class RegisterRecordAbility implements XMPushServiceListener {
    private final RegisterRecorder registerRecorder;

    public RegisterRecordAbility(RegisterRecorder registerRecorder) {
        this.registerRecorder = registerRecorder;
    }

    @Override
    public void start(Intent intent) {
        registerRecorder.recordRegisterRequest(intent);
    }
}

File: push/src/main/java/io/github/magisk317/mipush/service/RegisterRecordAbility.kt
package io.github.magisk317.mipush.service

import android.content.Intent

class RegisterRecordAbility(
    private val registerRecorder: RegisterRecorder
) : XMPushServiceListener {
    override fun start(intent: Intent) {
        registerRecorder.recordRegisterRequest(intent)
    }
}
